testing/trino-product-tests/src/main/java/io/trino/tests/product/utils/QueryExecutors.java

```diff
@@ -126,7 +126,7 @@ public static QueryExecutor onDelta()
 
         return new QueryExecutor()
         {
-            private final QueryExecutor delegate = testContext().getDependency(QueryExecutor.class, "delta");
+            private final QueryExecutor delegate = new DeltaQueryExecutor(testContext());
```

Comment:
`DeltaQueryExecutor` has a Connection open and is not part of the test context.
Who closes that connection?
are we leaking one connection  per every test that invokes `onDelta` ?